File: server/interfaces/conversion.go
package interfaces

import (
	"time"
)

// ConversionTask represents a conversion task
type ConversionTask struct {
	Path           string `json:"path"`
	ConversionType string `json:"conversion_type"`
	OutputPath     string `json:"output_path,omitempty"`
	TaskID         string `json:"task_id,omitempty"`
}

// ConversionResult represents the result of a conversion
type ConversionResult struct {
	Status         string    `json:"status"`
	ConversionType string    `json:"conversion_type"`
	Path           string    `json:"path"`
	StartTime      time.Time `json:"start_time"`
	EndTime        time.Time `json:"end_time,omitempty"`
	TotalFiles     int       `json:"total_files"`
	TotalSize      int64     `json:"total_size"`
	TotalConverted int       `json:"total_converted"`
	TotalFailed    int       `json:"total_failed"`
	ConvertedFiles []string  `json:"converted_files"`
	FailedFiles    []string  `json:"failed_files"`
	Error          string    `json:"error,omitempty"`
	TaskID         string    `json:"task_id,omitempty"`
}

// ActiveConversion represents an ongoing conversion
type ActiveConversion struct {
	StartTime time.Time      `json:"start_time"`
	Task      ConversionTask `json:"task"`
	Status    string         `json:"status"`
}

File: client/src/types/conversion.ts
export enum ConversionRoutes {
  FILE = "file", // Add FILE
  FOLDER = "folder",
  SETTINGS = "settings",
}

export enum ConversionTypes {
  EML = "EML",
  HTML = "HTML",
  PDF = "PDF",
  DOCX = "DOCX",
  DOC = "DOC",
  XLSX = "XLSX",
  XLS = "XLS",
  CSV = "CSV",
  JSON = "JSON",
  XML = "XML",
  TXT = "TXT",
  MD = "MD",
}

export interface ConversionSocketResponse {
  conversion_type: string;
  phase: number;
  files: string[];
  converted_files: number;
  failed_files: number;
  size: number;
}

export interface ConversionApiResponse {
  conversion_type: string;
  path: string;
  start_time: string;
  end_time: string;
  total_files: number;
  total_size: number;
  total_converted: number;
  total_failed: number;
  converted_files: string[];
  failed_files: string[];
}

export interface ConversionStartedEvent {
  conversion_type: string;
  path: string;
  start_time: string;
}

export interface PhaseStartedEvent {
  conversion_type: string;
  phase: number;
  total_files: number;
  total_size: number;
}

export interface FileProgressEvent {
  conversion_type: string;
  phase: number;
  current_file: string;
  success: boolean;
  converted_count: number;
  failed_count: number;
  total_files: number;
  error?: string;
}

export interface PhaseCompletedEvent {
  conversion_type: string;
  phase: number;
  converted_files: number;
  failed_files: number;
  total_files: number;
  size: number;
}

export interface ConversionCompletedEvent {
  conversion_type: string;
  path: string;
  start_time: string;
  end_time: string;
  total_files: number;
  total_size: number;
  total_converted: number;
  total_failed: number;
  converted_files: string[];
  failed_files: string[];
}

export interface ConversionErrorEvent {
  error: string;
  conversion_type: string;
  path: string;
}
